docs/public/specifications/tests/Basic-TP/basic01.md

```diff
@@ -89,169 +89,167 @@ DNS queries follow, unless otherwise specified below, what is specified for
-       then
-       1. Save the name server IP and parent zone name to the *Parent Found*
-          set.
-       2. Save the name server IP and parent zone name to the
-          *AA NXDomain Found* set.
-   3.  If the response has the AA flag set and an SOA record with
-       *Child Zone* as owner name in the answer section, then
-       1. If *Child Zone* is a [Direct Subdomain] to parent zone name then save the
-          name server IP and parent zone name to the *Parent Found* set.
-       2. Save the name server IP and parent zone name to the *AA SOA Found* set.
-   4.  If the response has the AA flag set and a CNAME record with
-       *Child Zone* as owner name in the answer section, then
-       1. Save the name server IP and parent zone name to the *Parent Found* set.
-       2. Save the name server IP and parent zone name to the *AA CNAME Found*
-          set.
-   5.  If the response is a [Referral] with a CNAME record with *Child Zone* as
-       owner name in the answer section, then
-       1. Save the name server IP and parent zone name to the *Parent Found* set.
-       2. Save the name server IP and parent zone name to the
-          *CNAME with Referral Found* set.
-   6.  If the response has the AA flag set and an empty answer section (NODATA),
-       then
-       1. [Send] a *DNAME Query* to the name server IP address.
-       2. If there is a response with the AA flag set, the [RCODE Name] NoError
-          and a DNAME record with *Child Zone* as owner name in the answer
-          section, then
-          1. Save the name server IP and parent zone name to the *Parent Found*
-            set.
-          2. Save the name server IP, parent zone name and the DNAME target
-            (RDATA value) to the *AA DNAME Found* set.
-       3. Else (no response or some other response than above), then do:
-          1. Save the name server IP and parent zone name to the *Parent Found*
-             set.
-          2. Save the name server IP and parent zone name to the
-             *AA NODATA Found* set.
-
-8. If the *Parent Found* set is non-empty, then
+
+   1.  Extract and remove *Server Address* including its *Zone Name* from
+       *Remaining Servers*.
+   2.  Insert *Server Address* and *Zone Name* into *Handled Servers*.
+   3.  Create [DNS queries][DNS Query]:
+       1. Query type SOA and query name *Zone Name* ("Zone Name SOA Query").
+       2. Query type NS and query name *Zone Name* ("Zone Name NS Query").
+   4.  [Send] *Zone Name SOA Query* to *Server Address*.
+   5.  Output [B01_SERVER_ZONE_ERROR] with query name *Zone Name*, [query type]
+       SOA and name server IP *Server Address* and go to next server in
+       *Remaining Servers* if one or more of the following matches:
+          * No DNS response.
+          * [RCODE Name] besides NoError in response.
+          * AA bit not set in response.
+          * No SOA record in answer section
```

Comment:
What if:

- there are two or more SOA records in the answer section?
- there are two SOA records, one whose owner name is (case-insensitively) equal to the QNAME and one whose owner name is different?